pyart/retrieve/echo_class.py

```diff
@@ -123,3 +127,354 @@ def steiner_conv_strat(grid, dx=None, dy=None, intense=42.0,
+
+    # parse the field parameters
+    if refl_field is None:
+        refl_field = get_field_name('reflectivity')
+    if zdr_field is None:
+        zdr_field = get_field_name('differential_reflectivity')
+    if rhv_field is None:
+        rhv_field = get_field_name('cross_correlation_ratio')
+    if kdp_field is None:
+        kdp_field = get_field_name('specific_differential_phase')
+    if temp_field is None:
+        temp_field = get_field_name('temperature')
+    if hydro_field is None:
+        hydro_field = get_field_name('radar_echo_classification')
+
+    # extract fields and parameters from radar
+    if refl_field in radar.fields:
+        refl = radar.fields[refl_field]['data']
+    else:
+        raise KeyError('Field not available: ' + refl_field)
+    if zdr_field in radar.fields:
+        zdr = radar.fields[zdr_field]['data']
+    else:
+        raise KeyError('Field not available: ' + zdr_field)
+    if rhv_field in radar.fields:
+        rhohv = radar.fields[rhv_field]['data']
+    else:
+        raise KeyError('Field not available: ' + rhv_field)
+    if kdp_field in radar.fields:
+        kdp = radar.fields[kdp_field]['data']
+    else:
+        raise KeyError('Field not available: ' + kdp_field)
+    if temp_field in radar.fields:
+        temp = radar.fields[temp_field]['data']
+    else:
+        raise KeyError('Field not available: ' + temp_field)
+
+    # convert temp in relative height respect to iso0
+    relh = temp*(1000./lapse_rate)
+
+    # data mask
+    # mask_zh = refl.mask
+
+    # standardize data
+    refl_std = _standardize(refl, 'Zh')
+    zdr_std = _standardize(zdr, 'ZDR')
+    kdp_std = _standardize(kdp, 'KDP')
+    rhohv_std = _standardize(rhohv, 'RhoHV')
+    relh_std = _standardize(relh, 'relH')
+
+    # standardize centroids
+    mc_std = np.zeros((nclasses, nvariables))
+    mc_std[:, 0] = _standardize(mass_centers[:, 0], 'Zh')
+    mc_std[:, 1] = _standardize(mass_centers[:, 1], 'ZDR')
+    mc_std[:, 2] = _standardize(mass_centers[:, 2], 'KDP')
+    mc_std[:, 3] = _standardize(mass_centers[:, 3], 'RhoHV')
+    mc_std[:, 4] = _standardize(mass_centers[:, 4], 'relH')
+
+    # assign to class
+    hydroclass_data, min_dist = _assign_to_class(
+        refl_std, zdr_std, kdp_std, rhohv_std, relh_std, mc_std,
+        weights=weights)
+
+    # prepare output fields
+    hydro = get_metadata(hydro_field)
+    hydro['data'] = hydroclass_data
+
+    return hydro
+
+
+def _standardize(data, field_name):
+    """
+    Streches the radar data to -1 to 1 interval
+
+    Parameters
+    ----------
+    data : array
+        radar field
+
+    field_name : str
+        type of field (relH, Zh, ZDR, KDP or RhoHV)
+
+    Returns
+    -------
+    field_std : dict
+        standardized radar data
+    """
+    if field_name == 'relH':
+        field_std = 2./(1.+np.ma.exp(-0.005*data))-1.
+        return field_std
+
+    if field_name == 'Zh':
+        mx = 60.
+        mn = -10.
```

Comment:
Hard coding these values may be problematic.  Would it make sense to have these in a lookup table?  A dictionary with field name keys and 2-tuple values which contain the min and max.  Then have this be an optional parameter which uses can supply but with these defaults?